Library/Homebrew/rubocops/lines.rb

```diff
@@ -665,6 +665,21 @@ def audit_formula(_node, _class_node, _parent_class_node, body_node)
         end
       end
 
+      # This cop ensures that new formulae depending on Requirements are not introduced in homebrew/core.
+      class CoreRequirements < FormulaCop
+        def audit_formula(_node, _class_node, _parent_class_node, _body_node)
+          if depends_on? :java
+            problem "Formulae in homebrew/core should depend on a versioned `openjdk` instead of :java"
```

Comment:
Should this each have allowlists until we're finished with the migration?

Reply:
Don't think so, it's an easy enough change.

Reply:
I guess these are strict audits so doesn't matter if they fail for now.

Reply:
Yeah, I figured it matters less since they're strict audits (and also general laziness). Can do it but I think it's more trouble than it's worth, especially since the java migration should be quite painless.